entrypoint/internal/utils/udev/udev.go

```diff
@@ -50,15 +51,76 @@ type udev struct {
 
 // CreateRules is the default implementation of the udev.Interface.
 func (u *udev) CreateRules(ctx context.Context) error {
-	_ = logr.FromContextOrDiscard(ctx)
-	// TODO add implementation
+	log := logr.FromContextOrDiscard(ctx)
+	log.Info("create udev rules")
+
+	// Create the udev rules content
+	// This creates rules to rename physical interfaces by removing the "np[n]" suffix
+	// Example: enp8s0f0np0 -> enp8s0f0, enp8s0f0np1v12 -> enp8s0f0v12
+	udevRulesContent := `ACTION!="add", GOTO="mlnx_ofed_name_end"
+SUBSYSTEM!="net", GOTO="mlnx_ofed_name_end"
+
+# Rename physical interfaces (first case) of virtual functions (second case).
+# Example names:
+# enp8s0f0np0 -> enp8s0f0
+# enp8s0f0np1v12 -> enp8s0f0v12
+
+DRIVERS=="mlx5_core", ENV{ID_NET_NAME_PATH}!="", \
+PROGRAM="/bin/sh -c 'echo $env{ID_NET_NAME_PATH} | sed -r -e s/np[01]$// -e s/np[01]v/v/'", \
+        ENV{ID_NET_NAME_PATH}="$result"
+
+DRIVERS=="mlx5_core", ENV{ID_NET_NAME_SLOT}!="", \
+PROGRAM="/bin/sh -c 'echo $env{ID_NET_NAME_SLOT} | sed -r -e s/np[01]$// -e s/np[01]v/v/'", \
+        ENV{ID_NET_NAME_SLOT}="$result"
+
+LABEL="mlnx_ofed_name_end"
```

Comment:
Do we want the rules file in the code or as a separate file in the repo?

Reply:
IMO, having a separate file for udev rules makes code more readable